### DISCUSSION TITLE
terminate called after throwing an instance of 'std::invalid_argument'

### DISCUSSION THREAD

**Discussion by hanfengsasa** [etcd-io/etcd#17053] [Q&A]
### Bug report criteria

- [ ] This bug report is not security related, security issues should be disclosed privately via security@etcd.io.
- [ ] This is not a support request or question, support requests or questions should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [ ] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [ ] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

terminate called after throwing an instance of 'std::invalid_argument'
  what():  stoi
SIGABRT: abort
PC=0x7f083d2a0387 m=3 sigcode=18446744073709551610

goroutine 0 [idle]:
runtime: unknown pc 0x7f083d2a0387
stack: frame={sp:0x7f0833ffe0f8, fp:0x0} stack=[0x7f08337ff268,0x7f0833ffee68)
00007f0833ffdff8:  0000000000000003  0000000000000000 
00007f0833ffe008:  0000000000000000  766e693a3a647473 
00007f0833ffe018:  6772615f64696c61  000000746e656d75 
00007f0833ffe028:  00007f0833ffea28  00007f0833ffe5b0 
00007f0833ffe038:  00007f0833ffe5b8  00007f0833ffee18 
00007f0833ffe048:  0000000000000000  00007f0833ffee20 
00007f0833ffe058:  000000000045faf0 <runtime.asmcgocall+112>  0000000000000000 
00007f0833ffe068:  0000000000000000  0000000000000000 
00007f0833ffe078:  00007f083cdd9a57  4000000000000000 
00007f0833ffe088:  0000000000000000  0000000000000000 
00007f0833ffe098:  0000000000000000  0000000000000000 
00007f0833ffe0a8:  0000000000000000  0000000000000000 
00007f0833ffe0b8:  0000000000000000  0000000000000000 
00007f0833ffe0c8:  00007f082c003f88  00007f083d632868 
00007f0833ffe0d8:  00007f082c004920  00007f0833ffe788 
00007f0833ffe0e8:  00007f0833ffe790  00007f0833ffe780 
00007f0833ffe0f8: <00007f083d2a1a78  0000000000000020 
00007f0833ffe108:  0000000000000000  0000000000000000 
00007f0833ffe118:  0000000000000000  0000000000000000 
00007f0833ffe128:  0000000000000000  0000000000000000 
00007f0833ffe138:  0000000000000000  0000000000000000 
00007f0833ffe148:  0000000000000000  0000000000000000 
00007f0833ffe158:  0000000000000000  0000000000000000 
00007f0833ffe168:  0000000000000000  0000000000000000 
00007f0833ffe178:  0000000000000000  00007f083d6321c0 
00007f0833ffe188:  00007f083d359ab9  0000000000000000 
00007f0833ffe198:  00007f083d2e42f3  00007f082c003f88 
00007f0833ffe1a8:  00007f083d6321c0  0000000000000001 
00007f0833ffe1b8:  00007f083d632243  00007f083d62e380 
00007f0833ffe1c8:  00007f083d2e5b0e  00007f083d6321c0 
00007f0833ffe1d8:  000000000000000a  00007f082c004920 
00007f0833ffe1e8:  00007f0833ffe788  00007f0833ffe790 
runtime: unknown pc 0x7f083d2a0387
stack: frame={sp:0x7f0833ffe0f8, fp:0x0} stack=[0x7f08337ff268,0x7f0833ffee68)
00007f0833ffdff8:  0000000000000003  0000000000000000 
00007f0833ffe008:  0000000000000000  766e693a3a647473 
00007f0833ffe018:  6772615f64696c61  000000746e656d75 
00007f0833ffe028:  00007f0833ffea28  00007f0833ffe5b0 
00007f0833ffe038:  00007f0833ffe5b8  00007f0833ffee18 
00007f0833ffe048:  0000000000000000  00007f0833ffee20 
00007f0833ffe058:  000000000045faf0 <runtime.asmcgocall+112>  0000000000000000 
00007f0833ffe068:  0000000000000000  0000000000000000 
00007f0833ffe078:  00007f083cdd9a57  4000000000000000 
00007f0833ffe088:  0000000000000000  0000000000000000 
00007f0833ffe098:  0000000000000000  0000000000000000 
00007f0833ffe0a8:  0000000000000000  0000000000000000 
00007f0833ffe0b8:  0000000000000000  0000000000000000 
00007f0833ffe0c8:  00007f082c003f88  00007f083d632868 
00007f0833ffe0d8:  00007f082c004920  00007f0833ffe788 
00007f0833ffe0e8:  00007f0833ffe790  00007f0833ffe780 
00007f0833ffe0f8: <00007f083d2a1a78  0000000000000020 
00007f0833ffe108:  0000000000000000  0000000000000000 
00007f0833ffe118:  0000000000000000  0000000000000000 
00007f0833ffe128:  0000000000000000  0000000000000000 
00007f0833ffe138:  0000000000000000  0000000000000000 
00007f0833ffe148:  0000000000000000  0000000000000000 
00007f0833ffe158:  0000000000000000  0000000000000000 
00007f0833ffe168:  0000000000000000  0000000000000000 
00007f0833ffe178:  0000000000000000  00007f083d6321c0 
00007f0833ffe188:  00007f083d359ab9  0000000000000000 
00007f0833ffe198:  00007f083d2e42f3  00007f082c003f88 
00007f0833ffe1a8:  00007f083d6321c0  0000000000000001 
00007f0833ffe1b8:  00007f083d632243  00007f083d62e380 
00007f0833ffe1c8:  00007f083d2e5b0e  00007f083d6321c0 
00007f0833ffe1d8:  000000000000000a  00007f082c004920 
00007f0833ffe1e8:  00007f0833ffe788  00007f0833ffe790 

goroutine 1142 [running]:
runtime.asmcgocall(0x7f083cdd9a57, 0xc001179158)
	/usr/local/go/src/runtime/asm_amd64.s:640 +0x42 fp=0xc001179140 sp=0xc001179138 pc=0x45fac2
runtime: unexpected return pc for debugCall64 called from 0x0
stack: frame={sp:0xc001179140, fp:0xc001179190} stack=[0xc001176000,0xc00117a000)
000000c001179040:  0000000000e95d7d  0000000000000000 
000000c001179050:  0000000000000000  000000c000180000 
000000c001179060:  0000000000000098  0000000000e95d20 
000000c001179070:  0000000000000008  0000000000000094 
000000c001179080:  0000000000000010  0000000000203000 
000000c001179090:  000000000000000f  000000c001179118 
000000c0011790a0:  0000000000203000  0000000000203000 
000000c0011790b0:  0000000000203000  0000000000203000 
000000c0011790c0:  0000000000203000  0000000000203000 
000000c0011790d0:  000000c000183b10  000000c000c7a2e8 
000000c0011790e0:  000000c001179180  000000000040de16 <runtime.mallocgc+790> 
000000c0011790f0:  000000c000183ca0  0000000000000010 
000000c001179100:  0000000000000008  0000000000e5a960 
000000c001179110:  0000000000000030  0000000000203000 
000000c001179120:  0000000000000008  0000000000000008 
000000c001179130:  000000000045fac2 <runtime.asmcgocall+66>  0000000000460780 <debugCall64+32> 
000000c001179140: <00007f083cdd9a57  000000c001179158 
000000c001179150:  00007f083db72008  0000000000015f33 
000000c001179160:  000000c000c1b800  000000c001179270 
000000c001179170:  000000c001179138  0000000000015f33 
000000c001179180:  000000c000b5c870 !0000000000000000 
000000c001179190: >0000000000000006  0000000000000048 
000000c0011791a0:  0000000000000048  00007f0838e64e44 
000000c0011791b0:  0000000000203000  0000000000000006 
000000c0011791c0:  00000000000000fa  00000000011812e6 
000000c0011791d0:  0000000000000000  000000000008ab64 
000000c0011791e0:  000000c001179140  0000000000000000 
000000c0011791f0:  00000000011aafc0  00000000011aafc0 
000000c001179200:  00000000011aaf80  6e656b6f7461227b 
000000c001179210:  2c22313030223a22  6369746f6e223a22 
000000c001179220:  3836312e3239313a  6570797472767322 
000000c001179230:  7465746164707522  39322d31312d3332 
000000c001179240:  73726576222c2231  3a22656e6f7a222c 
000000c001179250:  b7c5a71c54737a9c  0000000000000001 
000000c001179260:  08090a0b0c0d0e0f  000000c0011792f0 
000000c001179270:  0000000000b29fcf <go.etcd.io/etcd/api/v3/etcdserverpb.(*leaseClient).LeaseGrant+207>  000000c0000dca80 
000000c001179280:  00000000011b9a60  000000c000c413e0 
debugCall64()
	/usr/local/go/src/runtime/asm_amd64.s:1631 +0x20 fp=0xc001179190 sp=0xc001179140 pc=0x460780
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/var/lib/jenkins/workspace/p2p-new-cicd/p2pcloud/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:932 +0x204

### What did you expect to happen?

Program running

### How can we reproduce it (as minimally and precisely as possible)?

Why tune c++ libraries

### Etcd version (please run commands below)

<details>

```console
$ etcd --version
go.etcd.io/etcd/client/v3 v3.5.9
# paste output here

$ etcdctl version
# paste output here
```

</details>

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

**Comment by serathius** [comment]
Path including `/var/lib/jenkins/workspace/p2p-new-cicd/p2pcloud/pkg/` implies using non-official. We cannot be responsible debugging your own c++ libraries.

**Comment by hanfengsasa** [comment]
Maybe there is a problem with the following code, can you help me look at it

func (s *Stru_Etcd_Conn) SetValueWithTimeOut(key, value string, ttl int64) (int64, error) {
	key = s.SvrPathPrefix + key

	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
	defer cancel()

	lease := clientv3.NewLease(s.etcdcli)

	lresp, err := lease.Grant(ctx, ttl)

	if err != nil {
		return 0, err
	}

	kv := clientv3.NewKV(s.etcdcli)
	_, err = kv.Put(ctx, key, value, clientv3.WithLease(lresp.ID))

	return int64(lresp.ID), err
}

**Comment by tjungblu** [comment]
@hanfengsasa how can we repro this issue? Can you detail out how you run etcd (architecture, hardware, arguments to the binary)? 

As Marek said, it seems you're vendoring your own dependencies - so it would be useful for us to understand how you build it.
